app/bundles/CampaignBundle/EventListener/LeadSubscriber.php

```diff
@@ -72,19 +136,21 @@ public function onLeadListBatchChange(ListChangeEvent $event)
         $leads  = $event->getLeads();
         $list   = $event->getList();
         $action = $event->wasAdded() ? 'added' : 'removed';
-        $em     = $this->em;
 
         //get campaigns for the list
         if (!isset($listCampaigns[$list->getId()])) {
-            $listCampaigns[$list->getId()] = $this->campaignModel->getRepository()->getPublishedCampaignsByLeadLists($list->getId(), $this->security->isGranted('campaign:campaigns:viewother'));
+            $listCampaigns[$list->getId()] = $this->campaignModel->getRepository()->getPublishedCampaignsByLeadLists(
```

Comment:
I wanted to use the private property. But then some tests failed. I found out it's because model sets user to the repository before returning the repository. So I left it as it was. I will remove the `campaignRepository` property.

Reply:
Fixed in https://github.com/mautic/mautic/pull/8148/commits/1f762c8f4b5d85feeae9d4fc19ab6db5fc587aa5